test-benchmark/performance-test.html

```diff
@@ -0,0 +1,224 @@
+      }.bind(this));
+    },
+
+    /**
+     * True if at least one run has started.
+     *
+     * @type {boolean}
+     */
+    get started() {
+      return !!this.iframe.src;
+    },
+
+    /**
+     * True if all test runs have completed.
+     *
+     * @type {boolean}
+     */
+    get complete() {
+      return this.data.length === this.totalRuns;
+    },
+
+    /**
+     * The mean average of all collected data.
+     *
+     * @type {number}
+     */
+    get average() {
+      return this.data.length && this.data.reduce(function (sum, datum) {
+        return sum + datum.elapsed;
+      }, 0) / this.data.length;
+    },
+
+    /**
+     * The same as `average`, but fixed to two decimal places.
+     *
+     * @type {number}
+     */
+    get fixedAverage() {
+      return this.average.toFixed(2);
+    },
+
+    /**
+     * An array of numbers corresponding to the `elapsed` property of the
+     * collected data.
+     *
+     * @type {Array}
+     */
+    get series() {
+      return this.data.map(function (result) {
+        return result.elapsed;
+      });
+    },
+
+    /**
+     * The statistical variance of the `series`.
+     *
+     * @type {number}
+     */
+    get variance() {
+      var series = this.series;
+      var average = this.average;
+
+      return series.length && series.reduce(function (variance, value) {
+        return variance + Math.pow(value - average, 2);
+      }, 0) / series.length;
+    },
+
+    /**
+     * The standard deviation of the `series`.
+     *
+     * @type {number}
+     */
+    get standardDeviation() {
+      return Math.sqrt(this.variance);
+    },
+
+    /**
+     * The same as `standardDeviation`, but fixed to two decimal places.
+     *
+     * @type {number}
+     */
+    get fixedStandardDeviation() {
+      return this.standardDeviation.toFixed(2);
+    },
+
+    /**
+     * A comparison function suitable for usage with `Array.prototype.sort`.
+     * `PerformanceTest` instances are compared by `average`.
+     *
+     * @param {PerformanceTest} other A `PerformanceTest` instance to
+     * compare to this one.
+     * @return {number} 1 if this average is greater than other's average, -1
+     * if it is lesser, and 0 if they are equal.
+     */
+    compareTo: function(other) {
+      var average = this.average;
+      var otherAverage = other.average;
+
+      if (average > otherAverage) {
+        return 1;
+      } else if (average < otherAverage) {
+        return -1;
+      }
+
+      return 0;
+    },
+
+    /**
+     * Provides a comparative ratio of the averages of two tests.
+     *
+     * @param {PerformanceTest} other A `PerformanceTest` instance to compare
+     * @return {number} The ratio of the two test's averages.
+     */
+    averageRelativeTo: function(other) {
+      if (!other) {
+        return null;
+      }
+
+      var otherAverage = other.average;
+
+      return (this.average - otherAverage) / otherAverage;
```

Comment:
Yes! And in that case, we will helpfully report that the relative average is `Infinity` (or `0`, as the case may be).